ee/zentral/contrib/intune/api_client.py

```diff
@@ -0,0 +1,238 @@
+
+        self.add_ms_tree_extra_facts(ms_tree, device)
+        self.add_ms_tree_os_version(ms_tree, device)
+        self.add_ms_tree_disk(ms_tree, device)
+        self.add_ms_tree_network_interfaces(ms_tree, device)
+        self.add_ms_tree_principal_user(ms_tree, device)
+        return ms_tree
+
+    # helper methods
+
+    def get_business_unit_d(self):
+        return self.business_unit.serialize()
+
+    def get_source_d(self):
+        return {
+            "module": "zentral.contrib.intune",
+            "name": "Intune",
+            "config": {
+                "tenant_id": self.tenant_id
+            }
+        }
+
+    def add_ms_tree_extra_facts(self, ms_tree, device):
+        extra_facts = {}
+        compliance_status = device.compliance_state
+        if isinstance(compliance_status, str):
+            extra_facts["compliance_status"] = compliance_status
+        partner_reported_threat_state = self.add_ms_tree_extra_facts_partner_reported_threat_state(device)
+        if isinstance(partner_reported_threat_state, str):
+            extra_facts["partner_reported_threat_state"] = partner_reported_threat_state
+        extra_facts["azure_ad_device_id"] = device.azure_a_d_device_id
+        if extra_facts:
+            ms_tree["extra_facts"] = extra_facts
+
+    def add_ms_tree_extra_facts_partner_reported_threat_state(self, device):
+        match device.partner_reported_threat_state:
+            case 0:
+                return "Unknown"
+            case 1:
+                return "Activated"
+            case 2:
+                return "Deactivated"
+            case 3:
+                return "Secured"
+            case 4:
+                return "Low Severity"
+            case 5:
+                return "Medium Severity"
+            case 6:
+                return "High Severity"
+            case 7:
+                return "Unresponsive"
+            case 8:
+                return "Compromised"
+            case 9:
+                return "Misconfigured"
+        return None
+
+    def add_ms_tree_os_version(self, ms_tree, device):
+
+        def apple_version_tuple_to_parse(device):
+            try:
+                os_version_tuple = tuple(int(s) for s in device.os_version.split()[0].split('.'))
+                os_version = dict(zip(('major', 'minor', 'patch'), os_version_tuple))
+            except (TypeError, ValueError):
+                logger.warning("Device %s: could not parse OS version", device.id)
+                return
+            try:
+                os_build = device.os_version.split()[1].translate(str.maketrans("", "", "()"))
+                if os_build:
+                    os_version["build"] = os_build
+            except (TypeError, ValueError):
+                logger.warning("Device %s: could not parse build version", device.id)
+            return os_version_tuple, os_version
+
+        manufacturer = device.manufacturer
```

Comment:
I think it is better to use operating_system, and parse it as a windows os if == "Windows", then see if it is one of iOS iPadOS macOS or tvOS, and bail out if it is not something we know.

Also, it could be that an iPad is still reported as iOS, in that case, use the hardware model to see if it is an ipad and fix the os name if v > 13.1 (iPadOS from 13.1)